client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
> What is "this"
> What is "are for?"

Sorry, let me put it differently: Are the query managers not designed to manage Redux state? Sorry if they aren't, I saw it being used for posts and adopted it here. I'll think about how to do it differently then.